### PR TITLE
[student-libraries] Add ability to pre-load code in the js-interpreter

### PR DESCRIPTION
**Pull request by jmkulwik**
https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=26&projectKey=STAR&view=detail&selectedIssue=STAR-664

Pre-loads student library code while maintaining line numbers.

LibrariesWithLineNumbers

### CODE REVIEW

File: apps/src/lib/tools/jsinterpreter/CustomMarshalingInterpreter.js [hunk 1]
```diff
@@ -525,6 +525,7 @@ export default class CustomMarshalingInterpreter extends Interpreter {
    * @param {Function} intFunc The interpreter supplied callback function
    */
   marshalNativeToInterpreter(nativeVar, nativeParentObj, maxDepth) {
+    // debugger;
```
Comment (ajpal) [inline apps/src/lib/tools/jsinterpreter/CustomMarshalingInterpreter.js]: remove? And below

Comment (jmkulwik) [reply apps/src/lib/tools/jsinterpreter/CustomMarshalingInterpreter.js]: whoops... Fixed!

File: apps/src/applab/applab.js [hunk 2]
```diff
@@ -776,6 +778,30 @@ Applab.init = function(config) {
...
+    // config.dropletConfig.categories['libraryName'] = {
+    //   id: 'libraryName',
+    //   color: 'colorName',
+    //   rgb: 'colorHexCode',
+    //   blocks: []
+    // };
+
+    library.dropletConfig.forEach(dropletConfig => {
+      config.dropletConfig.blocks.push(dropletConfig);
+      level.codeFunctions[dropletConfig.func] = null;
```
Comment (ajpal) [inline apps/src/applab/applab.js:800]: why null?

**Review by ajpal — APPROVED** [review]

File: apps/src/applab/applab.js (hunk 2 above)
Comment (jmkulwik) [reply apps/src/applab/applab.js:800]: It's just the convention for adding a block to the toolbox:
https://github.com/code-dot-org/code-dot-org/blob/53f636676795c2bb9401326b6a108145747a1db2/lib/cdo/shared_constants.rb#L129

**Pull request merged**